// This is the Protobuf definition for RepoCommandResponse, described at:
// http://redmine.named-data.net/projects/repo-ng/wiki/Repo_Command
// Compile this file using:
// protoc --cpp_out=. repo-command-response.proto
syntax = "proto3";
package ndn_message;

// Protobuf has no "outer" message type, so we need to put the TLV RepoCommandResponse
// message inside an outer "typeless" RepoCommandResponseMessage.
message RepoCommandResponseMessage {
  message RepoCommandResponse {
    uint64 process_id = 206;
    uint64 status_code = 208;
    uint64 start_block_id = 204;
    uint64 end_block_id = 205;
    uint64 insert_num = 209;
    uint64 delete_num = 210;
  }

  // A RepoCommandResponseMessage has one RepoCommandResponse.
  RepoCommandResponse repo_command_response = 207;
}
